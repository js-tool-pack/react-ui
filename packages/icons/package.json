{
  "version": "0.0.2-beta.3",
  "types": "dist/index.d.ts",
  "main": "dist/index.js",
  "private": true,
  "scripts": {
    "rm:dist": "rimraf ./dist",
    "build:this": "tsc -p tsconfig.json",
    "build": "pnpm rm:dist && pnpm build:this",
    "check:ts": "tsc --noEmit -p tsconfig.json"
  },
  "devDependencies": {}
}
