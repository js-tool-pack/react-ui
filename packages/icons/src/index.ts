export * from './close.icon';
export * from './loading.icon';
export * from './circle-info.icon';
export * from './circle-info-fill.icon';
export * from './circle-success.icon';
export * from './circle-success-fill.icon';
export * from './circle-warning.icon';
export * from './circle-warning-fill.icon';
export * from './circle-close.icon';
export * from './circle-close-fill.icon';
export * from './up.icon';
export * from './right.icon';
export * from './down.icon';
export * from './left.icon';
export * from './selected.icon';
export * from './eye.icon';
export * from './eye-invisible.icon';
export * from './download.icon';
export * from './reset.icon';
export * from './zoom-in.icon';
export * from './zoom-out.icon';
export * from './rotate-left.icon';
export * from './rotate-right.icon';
export * from './flip-h.icon';
export * from './flip-v.icon';
export * from './empty.icon';
export * from './left-double.icon';
export * from './right-double.icon';
