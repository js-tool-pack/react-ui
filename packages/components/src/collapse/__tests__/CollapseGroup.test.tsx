import { CollapseGroupItem, CollapseGroup } from '..';
import { render } from '@testing-library/react';
import { Left } from '@pkg/icons';

describe('CollapseGroup', () => {
  test('all', () => {
    const nest: CollapseGroupItem[] = [
      {
        children:
          '希言自然。故飘风不终朝（zhāo），骤雨不终日。孰为此者？天地。天地尚不能久，而况于人乎？故从事于道者，道者同于道，德者同于德，失者同于失。同于道者，道亦乐得之；同于德者，德亦乐得之；同于失者，失亦乐得之。信不足焉，有不信焉。',
        icon: () => <Left />,
        iconPlacement: 'end',
        title: '第二十三章',
        key: '4',
      },
      {
        children:
          '企者不立，跨者不行，自见（xiàn）者不明，自是者不彰，自伐者无功，自矜者不长。其在道也，曰余食赘（zhuì）行。物或恶（wù）之，故有道者不处（chǔ）。',
        title: '第二十四章',
        disabled: true,
        key: '5',
      },
      {
        children:
          '有物混（hùn）成，先天地生。寂兮寥兮，独立不改，周行而不殆，可以为天下母。吾不知其名，字之曰道，强(qiǎng)为之名曰大。大曰逝，逝曰远，远曰反。故道大，天大，地大，王亦大。域中有四大，而王居其一焉。人法地，地法天，天法道，道法自然。',
        extra: <Left />,
        title: '第二十五章',
        key: '6',
      },
    ];
    const items: CollapseGroupItem[] = [
      {
        children:
          '孔德之容，惟道是从。道之为物，惟恍惟惚。惚兮恍兮，其中有象；恍兮惚兮，其中有物。窈（yǎo）兮冥兮，其中有精；其精甚真，其中有信。自古及今，其名不去，以阅众甫。吾何以知众甫之状哉？以此。',
        title: '第二十一章',
        key: '1',
      },
      {
        children:
          '曲则全，枉则直，洼则盈，敝则新，少则得，多则惑。是以圣人抱一，为天下式。不自见（xiàn）故明，不自是故彰，不自伐故有功，不自矜故长。夫唯不争，故天下莫能与之争。古之所谓曲则全者，岂虚言哉！诚全而归之。',
        title: '第二十二章',
        key: '2',
      },

      {
        children: (
          <CollapseGroup collapseProps={{ expanded: true }} items={nest} />
        ),
        title: '第二十三至二十五章',
        expanded: true,
        extra: '嵌套',
        key: '3',
      },
    ];
    expect(
      render(
        <CollapseGroup
          collapseProps={{ extra: '《道德经》', size: 'large' }}
          attrs={{ className: 'group' }}
          items={items}
          accordion
        />,
      ).container.firstChild,
    ).toMatchSnapshot();
  });
});
