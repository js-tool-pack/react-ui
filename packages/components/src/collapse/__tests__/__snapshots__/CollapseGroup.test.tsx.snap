// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollapseGroup all 1`] = `
<div
  class="t-collapse-group group"
>
  <section
    class="t-collapse t--size-lg"
    role="tab"
  >
    <div
      class="t-space t-collapse__header"
      style="--t-space-fill-ratio: 100%; gap: 12px;"
    >
      <i
        class="t-icon t-collapse__icon"
        role="img"
      >
        <svg
          viewBox="0 0 1024 1024"
        >
          <path
            d="M636.91389854 512.22982846L217.86080278 92.94690423l81.32081573-81.20590149 500.25899726 500.52713049L299.14331376 1012.25899726l-81.20590149-81.282511z"
            fill="currentColor"
          />
        </svg>
      </i>
      <div
        class="t-collapse__title"
      >
        第二十一章
      </div>
      《道德经》
    </div>
  </section>
  <section
    class="t-collapse t--size-lg"
    role="tab"
  >
    <div
      class="t-space t-collapse__header"
      style="--t-space-fill-ratio: 100%; gap: 12px;"
    >
      <i
        class="t-icon t-collapse__icon"
        role="img"
      >
        <svg
          viewBox="0 0 1024 1024"
        >
          <path
            d="M636.91389854 512.22982846L217.86080278 92.94690423l81.32081573-81.20590149 500.25899726 500.52713049L299.14331376 1012.25899726l-81.20590149-81.282511z"
            fill="currentColor"
          />
        </svg>
      </i>
      <div
        class="t-collapse__title"
      >
        第二十二章
      </div>
      《道德经》
    </div>
  </section>
  <section
    class="t-collapse t--size-lg t-collapse--active"
    role="tab"
  >
    <div
      class="t-space t-collapse__header"
      style="--t-space-fill-ratio: 100%; gap: 12px;"
    >
      <i
        class="t-icon t-collapse__icon t-collapse__icon--active"
        role="img"
      >
        <svg
          viewBox="0 0 1024 1024"
        >
          <path
            d="M636.91389854 512.22982846L217.86080278 92.94690423l81.32081573-81.20590149 500.25899726 500.52713049L299.14331376 1012.25899726l-81.20590149-81.282511z"
            fill="currentColor"
          />
        </svg>
      </i>
      <div
        class="t-collapse__title"
      >
        第二十三至二十五章
      </div>
      嵌套
    </div>
    <div
      class="t-collapse-transition t-collapse__content t-collapse-transition--h"
    >
      <div
        class="t-collapse__content-inner"
      >
        <div
          class="t-collapse-group"
        >
          <section
            class="t-collapse t--size-m t-collapse--active"
            role="tab"
          >
            <div
              class="t-space t-collapse__header"
              style="--t-space-fill-ratio: 100%; gap: 12px;"
            >
              <div
                class="t-collapse__title"
              >
                第二十三章
              </div>
              <svg
                viewBox="0 0 1024 1024"
              >
                <path
                  d="M808.12630286 92.94690423l-81.20590149-81.20590149-500.29730201 500.52713049 500.25899727 500.02916876 81.24420623-81.28251096-419.01479101-418.78496257z"
                  fill="currentColor"
                />
              </svg>
            </div>
            <div
              class="t-collapse-transition t-collapse__content t-collapse-transition--h"
            >
              <div
                class="t-collapse__content-inner"
              >
                希言自然。故飘风不终朝（zhāo），骤雨不终日。孰为此者？天地。天地尚不能久，而况于人乎？故从事于道者，道者同于道，德者同于德，失者同于失。同于道者，道亦乐得之；同于德者，德亦乐得之；同于失者，失亦乐得之。信不足焉，有不信焉。
              </div>
            </div>
          </section>
          <section
            class="t-collapse t--size-m t-collapse--disabled t-collapse--active"
            role="tab"
          >
            <div
              class="t-space t-collapse__header"
              style="--t-space-fill-ratio: 100%; gap: 12px;"
            >
              <i
                class="t-icon t-collapse__icon t-collapse__icon--active"
                role="img"
              >
                <svg
                  viewBox="0 0 1024 1024"
                >
                  <path
                    d="M636.91389854 512.22982846L217.86080278 92.94690423l81.32081573-81.20590149 500.25899726 500.52713049L299.14331376 1012.25899726l-81.20590149-81.282511z"
                    fill="currentColor"
                  />
                </svg>
              </i>
              <div
                class="t-collapse__title"
              >
                第二十四章
              </div>
            </div>
            <div
              class="t-collapse-transition t-collapse__content t-collapse-transition--h"
            >
              <div
                class="t-collapse__content-inner"
              >
                企者不立，跨者不行，自见（xiàn）者不明，自是者不彰，自伐者无功，自矜者不长。其在道也，曰余食赘（zhuì）行。物或恶（wù）之，故有道者不处（chǔ）。
              </div>
            </div>
          </section>
          <section
            class="t-collapse t--size-m t-collapse--active"
            role="tab"
          >
            <div
              class="t-space t-collapse__header"
              style="--t-space-fill-ratio: 100%; gap: 12px;"
            >
              <i
                class="t-icon t-collapse__icon t-collapse__icon--active"
                role="img"
              >
                <svg
                  viewBox="0 0 1024 1024"
                >
                  <path
                    d="M636.91389854 512.22982846L217.86080278 92.94690423l81.32081573-81.20590149 500.25899726 500.52713049L299.14331376 1012.25899726l-81.20590149-81.282511z"
                    fill="currentColor"
                  />
                </svg>
              </i>
              <div
                class="t-collapse__title"
              >
                第二十五章
              </div>
              <svg
                viewBox="0 0 1024 1024"
              >
                <path
                  d="M808.12630286 92.94690423l-81.20590149-81.20590149-500.29730201 500.52713049 500.25899727 500.02916876 81.24420623-81.28251096-419.01479101-418.78496257z"
                  fill="currentColor"
                />
              </svg>
            </div>
            <div
              class="t-collapse-transition t-collapse__content t-collapse-transition--h"
            >
              <div
                class="t-collapse__content-inner"
              >
                有物混（hùn）成，先天地生。寂兮寥兮，独立不改，周行而不殆，可以为天下母。吾不知其名，字之曰道，强(qiǎng)为之名曰大。大曰逝，逝曰远，远曰反。故道大，天大，地大，王亦大。域中有四大，而王居其一焉。人法地，地法天，天法道，道法自然。
              </div>
            </div>
          </section>
        </div>
      </div>
    </div>
  </section>
</div>
`;
